### PR TITLE
issue #4097 add warning when manually importing public keys

### LINKED ISSUE #4097: add warning when manually importing public keys
same as https://github.com/FlowCrypt/flowcrypt-android/issues/1544

### LINKED ISSUE #4104: The "This message includes a Public Key for ..." looks broken
Something is wrong here with whitespaces, it feels broken:

CleanShot 2021-11-09 at 15 53 52@2x

CleanShot 2021-11-09 at 15 55 11@2x

### PR DESCRIPTION
**Pull request by limonte**
This PR adds an orange warning when manually importing public keys. It also fixes the broken layout of `chrome/elements/pgp_pubkey.htm`

close #4097
close #4104

CleanShot 2021-11-10 at 18 26 08@2x

CleanShot 2021-11-10 at 18 24 26@2x

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]
The Bulk Import screenshot says "verify that the fingerprint matches" however, the fingerprint itself isn't displayed -- not very user-friendly.
Also, we have `add_pubkey.htm` dialog invoked from composer as I can see (example is demonstrated in `compose - settings - manually copied pubkey` test and some other tests). Apart from "Copy from Contact" options, there are also options to load from file or paste directly without any warnings whatsoever -- not even showing the email address inside the key/certificate. Perhaps, we should make use of `pgp_pubkey.htm` frame here somwehow -- file it in a separate issue?

**Comment by rrrooommmaaa** [comment]
Actually, the "Copy from Contact" option is looking ugly -- I'm filing an issue for that too

**Comment by tomholub** [comment]
Thank you for these observations.

Let's do the following in this PR:

> The Bulk Import screenshot says "verify that the fingerprint matches" however, the fingerprint itself isn't displayed -- not very user-friendly.

- [ ] Let's change the sentence in bulk import to: `Manually importing Public Keys can be dangerous`

> Also, we have `add_pubkey.htm` dialog invoked from composer as I can see (example is demonstrated in `compose - settings - manually copied pubkey` test and some other tests). Apart from "Copy from Contact" options, there are also options to load from file or paste directly without any warnings whatsoever

- [ ] `add_pubkey.htm` please add orange warning `Manually importing Public Keys can be dangerous`

Additionally in this PR:

- [ ] Please change the color of `Update Key` in `pgp_pubkey.htm` to orange
- [ ] similarly button color in bulk import to be orange
- [ ] the `OK` button in `add_pubkey.htm` also orange

That should cover it. Didn't notice this when writing the issue. Thanks!

**Comment by limonte** [comment]
> * [ ]  Let's change the sentence in bulk import to: `Manually importing Public Keys can be dangerous`

Done: 

CleanShot 2021-11-13 at 00 41 58@2x

> * [ ]  `add_pubkey.htm` please add orange warning `Manually importing Public Keys can be dangerous`

Done:

CleanShot 2021-11-13 at 00 41 16@2x

> * [ ]  Please change the color of `Update Key` in `pgp_pubkey.htm` to orange

Done:

CleanShot 2021-11-13 at 00 40 59@2x

> * [ ]  similarly button color in bulk import to be orange

Done, see screenshot above.

> * [ ]  the `OK` button in `add_pubkey.htm` also orange

Done, see screenshot above.

Thank you both for the detailed review and suggestions!

---

I personally have one UX concern. On both bulk import and `add_pubkey.htm` pages, we mention that "Manually importing Public Keys can be dangerous" without any further details answering "Why?" and "What to do about it?". Here's what GitHub co-pilot was suggesting:

CleanShot 2021-11-13 at 00 28 17@2x

It would be indeed helpful to create a small page with the information about importing and link that page on bulk import and `add_pubkey.htm` pages.

**Comment by tomholub** [comment]
Good idea about explaining it. I'll file an issue.

File: extension/chrome/settings/modules/contacts.ts [hunk 1]
```diff
@@ -265,7 +265,7 @@ View.run(class ContactsView extends View {
           }
         }
         container.css('display', 'block');
-        $('#bulk_import .input_pubkey, #bulk_import .action_process, #file_import #fineuploader_button').css('display', 'none');
+        $('#bulk_import .input_pubkey, #bulk_import .action_process, #file_import, #fineuploader_button').css('display', 'none');
```
Comment (limonte) [inline extension/chrome/settings/modules/contacts.ts:268]: This is the fix for the unrelated issue when the unwanted block was shown after doing bulk import:

CleanShot 2021-11-13 at 00 34 53@2x

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**